taxcalc/functions.py

```diff
@@ -1120,39 +1120,50 @@ def C1040(c05800, c07180, c07200, c07220, c07230, c07240, c07260, c07300,
     C1040 function computes total nonrefundable credits, c07100, and
                             income tax before refundable credits, c09200
     """
-    # total (nonrefundable) credits (2015 Form 1040, line 55)
+    # total nonrefundable credits (2015 Form 1040, line 55)
     c07100 = (c07180 + c07200 + c07600 + c07300 + c07400 + c07220 + c08000 +
               c07230 + c07240 + c07260)
     # tax after credits (2015 Form 1040, line 56)
-    nonrefundable_credits = max(0., c05800 - c07100)
+    tax_net_nonrefundable_credits = max(0., c05800 - c07100)
     # tax before refundable credits
     othertaxes = e09700 + e09800 + e09900 + NIIT
-    c09200 = othertaxes + nonrefundable_credits
+    c09200 = othertaxes + tax_net_nonrefundable_credits
     return (c07100, c09200)
 
 
 @iterate_jit(nopython=True)
-def IITAX(c09200, c59660, c11070, c10960, _eitc,
-          _payrolltax, personal_credit, n24, _iitax, _combined, _refund,
-          CTC_additional, CTC_additional_ps, CTC_additional_prt, c00100,
-          _sep, MARS):
-    """
-    IITAX function: ...
-    """
-    _refund = c59660 + c11070 + c10960 + personal_credit
+def IITAX(c59660, c11070, c10960, personal_credit,
+          c09200, _payrolltax,
+          CTC_new_c, CTC_new_rt,
+          n24, c00100, MARS,
+          CTC_new_ps, CTC_new_prt, CTC_new_refund_limit_rt,
+          ctc_new, _eitc, _refund, _iitax, _combined):
+    """
+    Compute final taxes including new refundable child tax credit
+    """
+    # compute new refundable child tax credit
+    if CTC_new_c > 0. and n24 > 0:
+        agi = max(0., c00100)
+        ctc = min(CTC_new_rt * agi, CTC_new_c)  # per kid credit
+        ymax = CTC_new_ps[MARS - 1]
+        if agi > ymax:
+            ctcx = max(0.,
+                       CTC_new_c - CTC_new_prt * (agi - ymax))
+            ctc = min(ctc, ctcx)
+        ctc_new = ctc * n24
+        if CTC_new_refund_limit_rt > 0. and ctc_new > 0.:
```

Comment:
But perhaps you'll say "someone will get an unexpected result if they set `CTC_new_refund_limit_rt` to a negative number for some reason [despite the fact that the validation key for the parameter constrains it to [0,1] interval]".  That line of thinking would imply this code:
```
        if ctc_new > 0.:
            refund_new = max(0., ctc_new - c09200)
            limit_new = CTC_new_refund_limit_rt * _payrolltax
            excess_new = max(0., refund_new - limit_new)
            ctc_new = max(0., ctc_new - excess_new)
```
So, `limit_new` would be negative, and hence, `excess_new > refund_new`.  For a filing unit with zero `c09200`, we would have `refund_new == ctc_new`.  So, the negative limit rate implies, in the last line, that that the "limited" ctc_new is **larger** than the original unlimited value of `ctc_new`.

Is that what someone putting in a negative value for `CTC_new_refund_limit_rt` would expect?  I can't imagine what someone would be thinking when they specified a negative value for the `CTC_new_refund_limit_rt` parameter.  What would they be trying to do?